torchvision/transforms/functional.py

```diff
@@ -179,6 +179,11 @@ def to_pil_image(pic, mode=None):
         if pic.ndimension() not in {2, 3}:
             raise ValueError('pic should be 2/3 dimensional. Got {} dimensions.'.format(pic.ndimension()))
 
+        elif pic.ndimension() == 3:
```

Comment:
Below we use unsqueeze to make pic as 3d tensor, so let's check and raise exception after the unsqueeze (which is not that expensive). Please, use also `pic.shape[-3]` instead of `pic.shape[0]` and no need to check `elif pic.ndimension() == 3`.

Reply:
This code can be removed ?